File: solutions/python/line-up/1/line_up.py
"""Line up sentence formatting."""

SUFFIXES = {
    1: "st",
    2: "nd",
    3: "rd",
}


def line_up(name, number):
    """Return a line-up message with the number as an ordinal."""
    return (
        f"{name}, you are the {number}{suffix_for(number)} "
        "customer we serve today. Thank you!"
    )


def suffix_for(number):
    """Return the ordinal suffix for a number."""
    if 11 <= number % 100 <= 13:
        return "th"

    return SUFFIXES.get(number % 10, "th")

File: solutions/javascript/line-up/1/line-up.js
const SUFFIXES = {
  1: 'st',
  2: 'nd',
  3: 'rd',
};

function suffixFor(number) {
  const lastTwoDigits = number % 100;
  if (lastTwoDigits >= 11 && lastTwoDigits <= 13) {
    return 'th';
  }

  return SUFFIXES[number % 10] ?? 'th';
}

export const format = (name, number) =>
  `${name}, you are the ${number}${suffixFor(number)} customer we serve today. Thank you!`;
